pallets/loans-ref/src/write_off.rs

```diff
@@ -0,0 +1,180 @@
+use cfg_primitives::Moment;
+use cfg_traits::ops::EnsureFixedPointNumber;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{
+	storage::bounded_btree_set::BoundedBTreeSet,
+	traits::tokens::{self},
+	RuntimeDebug,
+};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::Get, ArithmeticError, FixedPointNumber, FixedPointOperand};
+use sp_std::collections::btree_set::BTreeSet;
+use strum::EnumCount;
+
+/// Indicator of when the write off should be applied
+#[derive(
+	Encode,
+	Decode,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	TypeInfo,
+	RuntimeDebug,
+	MaxEncodedLen,
+	EnumCount,
```

Comment:
Why do we need that?

Reply:
It's used to bound the `triggers` vec (as the triggers need to be unique, thus the bound should be on the number of variants in this enum)